torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5134,7 +5134,7 @@ def reference_inputs_diagonal_diag_embed(op_info, device, dtype, requires_grad,
     samples3d = product(shapes3d, kwargs3d)
 
     for shape, kwargs in chain(samples1d, samples2d, samples3d):
-        if op_info.name in ('diagonal', '_refs.diagonal'):
+        if 'diagonal' in op_info.name:
```

Comment:
Nit: I don't like this name checks and would prefer we test for whether functions are equivalent instead:

```
if op.op in (torch.diagonal, ...)
```

Reply:
(Even better is to control this behavior through a kwarg on the function and have the OpInfos `partial` their sample inputs with the kwarg to control the behavior)

Reply:
I put this because otherwise I would have needed to put 4 different ops, which is a bit tiresome and a error-prone.